.github/workflows/nim-matrix.yml

```diff
@@ -6,7 +6,7 @@ on:
 
 env:
   cache_nonce: 0 # Allows for easily busting actions/cache caches
-  nim_version: v1.6.14, v1.6.16, v1.6.18
+  nim_version: pinned, v1.6.16, v1.6.18
```

Comment:
I think the latest version in nimbus-build-system is 1.16.21, we probably want to use that one instead of 1.6.16.

Reply:
also, where is pinned defined?

Reply:
Yeah sure, it's just that I think the current master won't even compile with 1.6.21. I'm going to bump it once we get the v4 merge in.

Reply:
The actual pin is [defined here](https://github.com/codex-storage/nim-codex/pull/832/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R18).

Reply:
Quick experiment on `master` (`check-nim-v1.6.20`)
<img width="1492" alt="Screenshot 2024-06-14 at 21 19 38" src="https://github.com/codex-storage/nim-codex/assets/20563034/fa06dc55-cd19-4f94-9f8b-86853b0801b0">

<img width="1161" alt="Screenshot 2024-06-14 at 21 19 06" src="https://github.com/codex-storage/nim-codex/assets/20563034/4c1a2442-544c-4324-a678-14bc814de636">

Reply:
OK good. :-) Anyhow, I'm going to bump it when we merge the v4 PR.

Reply:
Anyhow, this is somewhat beside the point: I'll bump it when we merge the Chronos v4 PR anyway or it won't work. This PR is about the pinning. :-)

Reply:
Just run a CI with v1.6.20 and looks like it works, but as you mentioned it is a different story

<img width="1491" alt="Screenshot 2024-06-14 at 23 56 27" src="https://github.com/codex-storage/nim-codex/assets/20563034/79e8ea97-82b9-4932-97f6-d3766d809c0f">